src/envoy/http/service_control/client_cache_test.cc

```diff
@@ -268,6 +283,184 @@ TEST_F(ClientCacheQuotaResponseErrorTypeTest, ApiKeyInvalid) {
   checkAndReset(stats_base_.stats().filter_.denied_consumer_error_, 1);
 }
 
+class ClientCacheHttpRequestTest : public ClientCacheTestBase {
+ public:
+  void SetUp() override {
+    service_config_.set_service_name(kServiceName);
+    service_config_.set_service_config_id(kServiceConfigId);
+
+    cache_ = std::make_unique<ClientCache>(
+        service_config_, filter_config_, stats_base_.stats(), cm_, time_source_,
+        dispatcher_, token_fn_, token_fn_);
+
+    // Setup mock http call.
+    http_call_ = std::make_unique<MockHttpCall>();
+    check_call_factory_ = std::make_unique<MockHttpCallFactory>();
+    quota_call_factory_ = std::make_unique<MockHttpCallFactory>();
+    report_call_factory_ = std::make_unique<MockHttpCallFactory>();
+  }
+
+  void TearDown() override {
+    EXPECT_EQ(got_num_callbacks, want_num_callbacks);
+    ClientCacheTestBase::TearDown();
+  }
+
+  void injectFactoryMocks() {
+    cache_->check_call_factory_ = std::move(check_call_factory_);
+    cache_->quota_call_factory_ = std::move(quota_call_factory_);
+    cache_->report_call_factory_ = std::move(report_call_factory_);
+  }
+
+  NiceMock<Envoy::Tracing::MockSpan> mock_parent_span_;
+  std::unique_ptr<MockHttpCall> http_call_;
+
+  // Ownership of these is passed to ClientCache after calling
+  // `injectFactoryMocks`.
+  std::unique_ptr<MockHttpCallFactory> check_call_factory_;
+  std::unique_ptr<MockHttpCallFactory> quota_call_factory_;
+  std::unique_ptr<MockHttpCallFactory> report_call_factory_;
+
+  // Track the number of callbacks. This will ensure no async calls are left on
+  // destruction / tearDown.
+  int got_num_callbacks = 0;
```

Comment:
per style guide add tailing _ for member variable

Reply:
Done.